ukat/mapping/phase_contrast.py

```diff
@@ -0,0 +1,221 @@
+        A prognostic marker in renal vascular diseases which range is [0, 1].
+    """
+
+    def __init__(self, velocity_array, affine, mask=None):
+        """Initialise a PhaseContrast class instance.
+
+        Parameters
+        ----------
+        velocity_array : np.ndarray
+            A 3D array containing the velocity images of the phase contrast
+            sequence, i.e. the dimensions of the array are [x, y, p], where p
+            corresponds to the phase (or trigger delay).
+        affine : np.ndarray
+            A matrix giving the relationship between voxel coordinates and
+            world coordinates.
+        mask : np.ndarray, optional
+            A boolean mask of the voxels to fit. Should be the shape of the
+            the raw data.
+        """
+        self.shape = velocity_array.shape
+        self.affine = affine
+        self.pixel_spacing = (np.linalg.norm(self.affine[:3, 1]),
+                              np.linalg.norm(self.affine[:3, 0]))
+        # Generate a mask if there isn't one specified
+        if mask is None:
+            self.mask = np.ones(self.shape, dtype=bool)
+        else:
+            self.mask = np.where(mask == 0, np.nan, mask)
+            # The purpose is for the np.nanmean to work properly later
+            # without including the 0s of the mask
+        self.velocity_array = np.abs(velocity_array * self.mask)
+        self.num_pixels_phase = []
+        self.area_phase = []
+        self.min_velocity_phase = []
+        self.mean_velocity_phase = []
+        self.max_velocity_phase = []
+        self.peak_velocity_phase = []
+        self.std_velocity_phase = []
+        self.RBF = []
+        self.mean_velocity = 0
+        self.mean_RBF = 0
+        self.resistive_index = 0
+        if len(self.shape) == 3:
+            for phase in range(self.shape[-1]):
+                phase_array = self.velocity_array[..., phase]
+                num_pixels = np.count_nonzero(~np.isnan(phase_array))
+                area = (num_pixels * 0.1 * self.pixel_spacing[0] *
+                        0.1 * self.pixel_spacing[1])  # (0.1*mm * 0.1*mm) = cm2
+                min_vel = np.nanmin(phase_array)
+                avrg_vel = np.nanmean(phase_array)
+                max_vel = np.nanmax(phase_array)
+                std_vel = np.nanstd(phase_array)
+                Q = 60 * area * avrg_vel  # (60s*cm2*cm/s) = cm3/min = ml/min
+                self.num_pixels_phase.append(num_pixels)
+                self.area_phase.append(area)
+                self.min_velocity_phase.append(min_vel)
+                self.mean_velocity_phase.append(avrg_vel)
+                self.max_velocity_phase.append(max_vel)
+                self.peak_velocity_phase.append(max_vel)
+                self.std_velocity_phase.append(std_vel)
+                self.RBF.append(Q)
+            # Build table with the results per phase (list of lists)
+            self.stats_table = {"Phase": list(np.arange(self.shape[-1])),
```

Comment:
Could this be renamed to stats_dict (or even just stats). To me table could mean many things but a dictionary is never one of them.